airflow/ui/src/components/AppContainer/TimezoneDropdown.tsx

```diff
@@ -0,0 +1,90 @@
+/*!
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React, { useState } from 'react';
+import {
+  Box,
+  Button,
+  Menu,
+  MenuButton,
+  MenuList,
+  Tooltip,
+} from '@chakra-ui/react';
+import dayjs from 'dayjs';
+import tz from 'dayjs/plugin/timezone';
+import Select from 'components/MultiSelect';
+
+import timezones from 'utils/timezones.json';
+import { useTimezoneContext } from 'providers/TimezoneProvider';
+
+dayjs.extend(tz);
+
+interface Option { value: string, label: string }
+
+const TimezoneDropdown: React.FC = () => {
+  const { timezone, setTimezone } = useTimezoneContext();
+  const [now, setNow] = useState(dayjs().tz(timezone));
+
+  let currentTimezone;
+  timezones.find((group) => (
+    group.options.find((zone) => {
+      if (zone.value === timezone) {
+        currentTimezone = zone;
+        return true;
+      }
+      return false;
+    })
+  ));
+
+  const onChangeTimezone = (newTimezone: Option | null) => {
+    if (newTimezone) {
+      setTimezone(newTimezone.value);
+      setNow(dayjs().tz(newTimezone.value));
+    }
+  };
+
+  return (
+    <Menu isLazy>
+      <Tooltip label="Change time zone" hasArrow>
+        <MenuButton as={Button} variant="ghost" mr="4">
+          <Box
+            as="time"
+            dateTime={now.toString()}
+            fontSize="md"
+          >
+            {now.format('h:mmA Z')}
```

Comment:
So here's the thing about localization: the "common" way dates are displayed in about half the world is to use the 24 hour clock

https://en.wikipedia.org/wiki/Date_and_time_representation_by_country#/media/File:12_24_Hours_World_Map.svg

This is possibly an extra feature/PR as I guess the time display is a per-user setting, not a property of what timezone is selected.